From 94f416226aa7f3580cb2a9accf61a84317aec9db Mon Sep 17 00:00:00 2001
From: Balaji Eswaramurthy <balaji.eswaramurthy@qubercomm.com>
Date: Thu, 7 Nov 2019 18:22:45 +0530
Subject: [PATCH] Nmeshd: Code optimization

 Prevoiusly the for loop is running every time inside the loop.
 Now it will run when the json key mathes with bssid.

 if condition statements are changed to else if condition.
---
 cloudapd/cloudiface/json.c | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/cloudapd/cloudiface/json.c b/cloudapd/cloudiface/json.c
index 7be47ed..5254af1 100644
--- a/cloudapd/cloudiface/json.c
+++ b/cloudapd/cloudiface/json.c
@@ -1361,24 +1361,24 @@ static void json_mesh_parse_rrm_resp_status(struct capd *capd,
 			index = sta->rrm_resp->nresp;
 			json_object_object_get_ex(info_obj, key, &info_attr);
 
-			if (!strcmp(key, BSSID))
+			if (!strcmp(key, BSSID)) {
 				mac_aton((char *)json_object_get_string(info_attr),
 					 bssid);
 
-			for (j = 0; j < sta->rrm_resp->nresp; j++) {
-				if (!memcmp(&bssid, &sta->rrm_resp->resp_info[j].bssid, ETH_ALEN)) {
-					index = j;
-					break;
+				for (j = 0; j < sta->rrm_resp->nresp; j++) {
+					if (!memcmp(&bssid, &sta->rrm_resp->resp_info[j].bssid,
+						    ETH_ALEN)) {
+						index = j;
+						break;
+					}
 				}
-			}
-
-			memcpy(sta->rrm_resp->resp_info[index].bssid, bssid, ETH_ALEN);
 
-			if (!strcmp(key, RCPI))
+				memcpy(sta->rrm_resp->resp_info[index].bssid, bssid, ETH_ALEN);
+			} else if (!strcmp(key, RCPI))
 				sta->rrm_resp->resp_info[index].rcpi =
 					json_object_get_int(info_attr);
 
-			if (!strcmp(key, RSNI))
+			else if (!strcmp(key, RSNI))
 				sta->rrm_resp->resp_info[index].rsni =
 					json_object_get_int(info_attr);
 
-- 
2.7.4

